BUMP_VERSION.asciidoc

```diff
@@ -0,0 +1,196 @@
+= How to bump the Elasticsearch version
+
+[partintro]
+
+This is a short guide on all the steps required to bump the Elasticsearch
+version number to the next version after a release has been cut.
+
+== Introduction
+
+In order to make the steps concrete, it helps to work through an example.  
+In this scenario, suppose our current major release version of Elasticsearch
+is version 5, and Elasticsearch's current version of 5.x is 5.2.0.  The 
+next major release version is 6, which is reflected in the master branch.  
+
+== Steps to take for feature freeze
+
+When ready to feature freeze 5.2.0, the following steps must be taken.
+
+==== Create the branch
+
+Create a new `5.2` branch from the `5.x` branch and push it to Github.  
+The `5.2` branch will represent the 5.2 line of 5.x.
+-------------------------------------
+git checkout -b 5.2 upstream/5.x
+git push upstream 5.2
```

Comment:
@clintongormley if you execute the `git checkout -b 5.2 upstream/5.x`, it automatically sets up remote tracking and git pull works (at least for me).  Is it different for you?